src/spec.rs

```diff
@@ -423,8 +414,17 @@ impl Zip64CentralDirectoryEnd {
 
                 results.push((cde, archive_offset));
             }
+            if window_start == nominal_offset {
+                break;
+            }
+            debug_assert!(
+                END_WINDOW_SIZE > mem::size_of_val(&ZIP64_CENTRAL_DIRECTORY_END_SIGNATURE)
+            );
             window_start = window_start
-                .saturating_sub(END_WINDOW_SIZE as u64)
+                .saturating_sub(
+                    END_WINDOW_SIZE as u64
```

Comment:
I'd actually add the overlap before doing the `saturating_sub`.

Reply:
Done!